### PR TITLE
Developer Landing Page

### PR DESCRIPTION
**Pull request by bencgreenberg**
As an iterative step towards a more comprehensive developer portal, a developer landing page will provide a focused place where all the various developer-related resources can be found in a consolidated listing

### CODE REVIEW

**Comment by phacks** [comment]
Great job on this @bencgreenberg!

## Content

To really set that page apart from the rest of the website and get that “developer-y” feeling more pronounce, we could focus on developer-friendly content first:

- Remove the Getting started and Integrations sections as those are more about the product than the platform;
- Split the Community section in two: “Join the community” with a link to the Discord so that folks know that they can reach out to other builders and ask questions, and “Built by the community” with links to open source projects built by the community. Build by the community could be among the first blocks as I find it a very strong signal as a developer, and a good thing to browse and get ideas from.
- Change “Beta Features” to “Join the Beta program” to switch from passive voice to the active one. Can also mention that the beta program is a great way to suggest improvements to features/API before they reach GA.
- That would require deeper changes (although totally feasible) but if we had tags on the blog we could link to our “Engineering” blog directly, i.e. we filter down by the “Engineering” tag.
- We have a few engineering-focused talks by the Orbit team that could maybe find a home in this page as well? Yours as Rails Conf, mines at the Eleventies and Jamstack Berlin… Not 100% sure this is a good idea, just throwing it out there.

## Design

- Wondering whether we should remove the Orbit Model banner at the bottom as it’s standing out a bit too much currently
- I riffed on an alternative to the content on each card:

image

Adding a few words of description with each item can let us be more explicit, or creative, or geeky etc…

<details>
<summary>Here’s the markup if you want to reproduce</summary>

```html
<ul class="text-left">
  <li class="mb-2">
    <dl class="font-medium text-purple-400"><a href="#">API reference</a></dl>
    <dt class="text-gray-500 pl-4 mt-1">View and test all endpoints in the browser</dt>
  </li><li class="mb-2">
    <dl class="font-medium text-purple-400">Key concepts</dl>
    <dt class="text-gray-500 pl-4 mt-1">Learn about Orbit concepts and primitives</dt>
  </li><li class="mb-2">
    <dl class="font-medium text-purple-400">Common API calls</dl>
    <dt class="text-gray-500 pl-4 mt-1">Handy code snippets for the most common requests</dt>
  </li><li>
    <dl class="font-medium text-purple-400">Swagger/Postman definitions</dl>
    <dt class="text-gray-500 pl-4 mt-1">Import the API reference into your favourite tool</dt>
  </li>
</ul>
```

</details>

**Comment by bencgreenberg** [comment]
@phacks, thank you for such a thoughtful review. I don't think I've been greeted before by such deliberateness in consideration and in the formatting of it! ❤️ 

Let me respond to your points:

## Content

> Remove the Getting started and Integrations sections as those are more about the product than the platform;

I'm going to keep the getting started section, as information on the product is also integral, but I believe you are right about the integrations section. What I would like eventually is a showcase of our own eventual developer tooling to _enable_ a developer to build their own integrations, but the integrations page is not that.

> Split the Community section in two: “Join the community” with a link to the Discord so that folks know that they can reach out to other builders and ask questions, and “Built by the community” with links to open source projects built by the community. Build by the community could be among the first blocks as I find it a very strong signal as a developer, and a good thing to browse and get ideas from.

This is a fabulous idea! I'll integrate it as such.

> Change “Beta Features” to “Join the Beta program” to switch from passive voice to the active one. Can also mention that the beta program is a great way to suggest improvements to features/API before they reach GA.

An active positioning means we need to have a link to join. Is it the link to the Airtable form? Ideally, we'd also have a landing page for the beta program that enumerated its benefits that this would link to, instead of directly linking to the form. TBC for later.

> That would require deeper changes (although totally feasible) but if we had tags on the blog we could link to our “Engineering” blog directly, i.e. we filter down by the “Engineering” tag.

We do tag blog posts on Sanity. I've been using a "Developer Relations" one for example. Can we filter by those? I'm guessing that data is exposed somehow to our site and we could create a blog view based on tag?

> We have a few engineering-focused talks by the Orbit team that could maybe find a home in this page as well? Yours as Rails Conf, mines at the Eleventies and Jamstack Berlin… Not 100% sure this is a good idea, just throwing it out there.

Yes! An entire area devoted to our presence in communities (meetups, talks, etc.) is a must-have. Consider that a coming soon feature. 

## Design

> Wondering whether we should remove the Orbit Model banner at the bottom as it’s standing out a bit too much currently

I was feeling the same way. I do like having some call out to the conceptual work behind Orbit. I'd like for it to be minimal though as a developer page should focus on the practical, nonetheless, a small mention somewhere if someone wanted to read up on it would be good. Can be incorporated later.

> I riffed on an alternative to the content on each card:

Much better! 👏🏻

**Comment by phacks** [comment]
Love how it turned out @bencgreenberg! Thanks for the amazingly quick turnaround on the suggestions. Couple more ideas:

- For links inside card subheaders, use `underline` instead of `font-purple-400` so that they feel “less important” than the subheaders links (top is my suggestion, bottom is the current look & feel):

image

- The current contrast ratio of card description falls short of being accessible given the background color. Changing from `font-gray-500` to `font-gray-600` does the trick. In the following screenshots I change the “_ Explore and utilize open-source tooling built_” bit.

Before:

image

After:

image

<details>
  <summary>Here’s a video on how I tested the contrast ratio using the dev tools</summary>
  https://user-images.githubusercontent.com/2587348/115002104-a5e4ea00-9ea4-11eb-8957-a90b81754409.mp4
</details>

- Should Getting Started > Integrations be renamed Getting Started > GitHub Integration?
- Orbit API > Batch/Bulk Data Import is also in the Community Built section, we might want to only have it there
- Node-RED could link to https://nodered.org/
- I think Kevin built another open source project for in-person events, should we add it in the Community Built section?
- In the Beta section, Alerts & Destinations and Webhooks are _nearly_ the same thing (i.e. Webhooks is the big thing powered by Alerts & Destinations). We might want to only keep Webhook for conciseness.
- Should we add Community Built to the Join the Community card?

**Comment by phacks** [comment]
> Can we filter by those? I'm guessing that data is exposed somehow to our site and we could create a blog view based on tag?

I think everything is set up in Sanity, but we’re missing the filtering code in the website. As it’s a fairly common use-case for Eleventy I don’t anticipate it being that difficult to implement though.

**Comment by phacks** [comment]
Once we’re confident in the content and the design, there’s a code refactoring that I have in mind that might help create a reusable component for those cards and make future changes simpler. We can leverage Nunjucks’ `macros` to create a `card` macro which will take its content as parameters. We’re using that patterns in a few places, most notably in the integrations page.

Here we create a new `integration` macro:

https://github.com/orbit-love/orbit-web/blob/d656aa888aa6fea18bfe352df5587d11fcf630dc/src/includes/macros/features.njk#L92-L124

Here we instantiate it for e.g. the Discourse integration:

https://github.com/orbit-love/orbit-web/blob/d656aa888aa6fea18bfe352df5587d11fcf630dc/src/integrations.html#L45-L67

Happy to pair on this if you’d like!

**Comment by bencgreenberg** [comment]
>  But as the website is going to change, I’m not sure yet if that’s worth the work. We can talk about that with Gareth!

That's a very good point. We can hold off.

**Comment by bencgreenberg** [comment]
> Once we’re confident in the content and the design, there’s a code refactoring that I have in mind that might help create a reusable component for those cards and make future changes simpler. We can leverage Nunjucks’ `macros` to create a `card` macro which will take its content as parameters. We’re using that patterns in a few places, most notably in the integrations page.
> 
> Here we create a new `integration` macro:
> 
> https://github.com/orbit-love/orbit-web/blob/d656aa888aa6fea18bfe352df5587d11fcf630dc/src/includes/macros/features.njk#L92-L124
> 
> Here we instantiate it for e.g. the Discourse integration:
> 
> https://github.com/orbit-love/orbit-web/blob/d656aa888aa6fea18bfe352df5587d11fcf630dc/src/integrations.html#L45-L67
> 
> Happy to pair on this if you’d like!

Ah, I just saw this comment now. Yes, a reusable component looks like a good idea. The templating looks similar to Rails templates. I think I can work on this.

**Comment by phacks** [comment]
> I think I can work on this.

Awesome. Let me know if I can help!

Out of curiosity I tried swapping the third and fourth cards, and it mitigates a bit the whitespace caused by the long “Community Built” card. Let me know what you think!

image

File: src/includes/macros/developers/cards.njk [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{% macro card(content) %}
+<div class="pt-6">
+    <div class="flow-root bg-gray-100 rounded-lg px-6 pb-8">
+    <div class="-mt-6">
+        <div>
```
Comment (phacks) [inline src/includes/macros/developers/cards.njk:5]: Might be wrong, but I think that `<div>` wrapping the `<span>` can be safely removed

File: src/includes/footer.njk [hunk 2]
```diff
@@ -7,11 +7,10 @@
       <a href="/blog/" class="header_link__desktop mb-2">Blog</a>
       <a href="/podcast/" class="header_link__desktop mb-2">Podcast</a>
       <a
-        href="https://docs.orbit.love/docs"
-        target="_blank"
+        href="https://docs.orbit.love/developers"
```
Comment (phacks) [inline src/includes/footer.njk:10]: shouldn’t that be https://orbit.love/developers?

File: src/includes/developers/offerings.njk [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+            },
+            {
+              title: "Custom Integrations Node.js App",
+              link: "https://github.com/nexmo/enter-our-orbit",
+              description: "Integrate Twitch, YouTube and Stack Overflow interacts with this Node.js app by the <a href='https://twitter.com/vonagedev' class='underline'>Vonage Developer Relations Team</a>"
+            },
+            {
+              title: "DEV to Orbit Ruby Gem",
+              link: "https://github.com/bencgreenberg/dev_orbit",
+              description: "ntegrate DEV Community interactions into your Orbit workspace in Ruby by <a href='https://twitter.com/rabbigreenberg' class='underline'>Ben Greenberg</a>"
```
Comment (phacks) [inline src/includes/developers/offerings.njk:147]: Missing an `I` there!

File: src/includes/footer.njk (hunk 2 above)
Comment (bencgreenberg) [reply src/includes/footer.njk]: 👍🏻

File: src/includes/macros/developers/cards.njk [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        <span class="inline-flex items-center justify-center p-3 bg-purple-400 rounded-md shadow-lg">
+            <svg class="h-6 w-6 text-white" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">
+                {{ content.svg | safe }}
+            </svg>
+        </span>
+        </div>
+        <h3 class="mt-8 text-lg font-medium text-gray-900 tracking-tight">{{ content.title }}</h3>
+        <p class="mt-5 text-base text-gray-500">
```
Comment (phacks) [inline src/includes/macros/developers/cards.njk:13]:
```suggestion
        <p class="mt-5 text-base text-gray-600">
```

File: src/includes/macros/developers/cards.njk [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        {{ content.description }}
+        </p>
+
+        <hr class="mt-4 mb-4 mx-auto" />
+
+        <ul class="text-left">
+        {% for section in content.sections %}
+            <li class="mb-2">
+                <dl class="font-medium text-purple-400"><a href={{ section.link }} class="focus:outline-none">{{ section.title }}</a></dl>
+                <dt class="text-gray-500 pl-4 mt-1">{{ section.description | safe }}</dt>
```
Comment (phacks) [inline src/includes/macros/developers/cards.njk:23]:
```suggestion
                <dt class="text-gray-600 pl-4 mt-1">{{ section.description | safe }}</dt>
```

**Comment by phacks** [comment]
Apart from the few tiny comments/typos, I think this is ready to merge! Feel free to hit that button when you’re ready, it should be deployed about a minute after

File: src/includes/developers/offerings.njk [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+            }
+          ]
+        }
+      %}
+      {{ card(orbit_api) }}
+
+      {%
+        set beta_program = {
+          title: "Join the Beta Program",
+          description: "Get early access to beta features in development and become a part of the community offering product feedback and improvements.",
```
Comment (phacks) [inline src/includes/developers/offerings.njk]:
```suggestion
          description: "Get early access to beta features and become a part of the community offering product feedback and improvements.",
```

“beta features in development” sounds a bit redundant

File: src/includes/developers/offerings.njk [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+            },
+            {
+              title: "Slack to Orbit Zapier Template #1",
+              link: "https://zapier.com/shared/abe6fd30302965cdaba5ac3d60c8bc19ea1ae791",
+              description: "Format Slack messages and send them to Orbit on Zapier with this template by <a href='https://twitter.com/mikermcneil' class='underline'>Mike NcNeil</a>"
+            },
+            {
+              title: "Slack to Orbit Zapier Template #2",
+              link: "https://zapier.com/shared/e3bd8574eceb8a7f53608394e5926d6952107a1e",
+              description: "Capture Slack anywhere messages and send them to Orbit as a new activity by <a href='https://twitter.com/fharper' class='underline'>Frédéric Harper</a>"
```
Comment (phacks) [inline src/includes/developers/offerings.njk:132]:
```suggestion
              description: "Capture Slack messages and send them to Orbit as a new activity by <a href='https://twitter.com/fharper' class='underline'>Frédéric Harper</a>"
```

File: src/includes/developers/offerings.njk [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+            },
+            {
+              title: "Batch/Bulk Data Import",
+              link: "https://airtable.com/universe/expZVjdIZgZpw2dK2/orbit-batchbulk-data-import",
+              description: "Airtable template and script for bulk importing historical activities by <a href='https://twitter.com/_phzn' class='underline'>Kevin Lewis</a>"
+            },
+            {
+              title: "Custom Integrations Node.js App",
+              link: "https://github.com/nexmo/enter-our-orbit",
+              description: "Integrate Twitch, YouTube and Stack Overflow interacts with this Node.js app by the <a href='https://twitter.com/vonagedev' class='underline'>Vonage Developer Relations Team</a>"
```
Comment (phacks) [inline src/includes/developers/offerings.njk:142]:
```suggestion
              description: "Integrate Twitch, YouTube, and Stack Overflow with this Node.js app by the <a href='https://twitter.com/vonagedev' class='underline'>Vonage Developer Relations Team</a>"
```

File: src/includes/developers/offerings.njk [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+              link: "https://github.com/davidgs/OrbitParser",
+              description: "Parse your Orbit JSON into a CSV file in Golang by <a href='https://twitter.com/davidgsIoT' class='underline'>David G. Simmons</a>"
+            },
+            {
+              title: "Slack to Orbit Zapier Template #1",
+              link: "https://zapier.com/shared/abe6fd30302965cdaba5ac3d60c8bc19ea1ae791",
+              description: "Format Slack messages and send them to Orbit on Zapier with this template by <a href='https://twitter.com/mikermcneil' class='underline'>Mike NcNeil</a>"
+            },
+            {
+              title: "Slack to Orbit Zapier Template #2",
```
Comment (phacks) [inline src/includes/developers/offerings.njk:130]: Open question: those two Slack to Orbit templates are now obsolete, as Slack is now a third party integration. Should we include them as valuable community-built resource, or would that be too confusing?

File: src/includes/developers/offerings.njk (hunk 7 above)
Comment (bencgreenberg) [reply src/includes/developers/offerings.njk:132]: In that case that's a copy and paste from the resources page ;)

File: src/includes/developers/offerings.njk (hunk 9 above)
Comment (bencgreenberg) [reply src/includes/developers/offerings.njk:130]: That's a very good point. Given that, I'd say we can safely remove them from the list. We should also remove them from the community resources page, from which this list was generated. I'll do that early next week.

**Pull request merged**